deployment/terraform/api.tf

```diff
@@ -0,0 +1,143 @@
+#
+# Security Group Resources
+#
+resource "aws_security_group" "alb" {
+  name   = "sg${var.project}APILoadBalancer"
+  vpc_id = data.terraform_remote_state.core.outputs.vpc_id
+
+  tags = {
+    Name    = "sg${var.project}APILoadBalancer",
+    Project = var.project
+  }
+}
+
+#
+# ALB Resources
+#
+resource "aws_lb" "api" {
+  name            = "alb${var.project}API"
+  security_groups = [aws_security_group.alb.id]
+  subnets         = data.terraform_remote_state.core.outputs.public_subnet_ids
+
+  enable_http2 = true
+
+  tags = {
+    Name    = "alb${var.project}API"
+    Project = var.project
+  }
+}
+
+resource "aws_lb_target_group" "api" {
+  name = "tg${var.project}API"
+
+  health_check {
+    healthy_threshold   = 3
+    interval            = 30
+    matcher             = 200
+    protocol            = "HTTP"
+    timeout             = 3
+    path                = "/api/hello/world"
+    unhealthy_threshold = 2
+  }
+
+  port     = 80
+  protocol = "HTTP"
+  vpc_id   = data.terraform_remote_state.core.outputs.vpc_id
+
+  tags = {
+    Name    = "tg${var.project}API"
+    Project = var.project
+  }
+}
+
+resource "aws_lb_listener" "api_redirect" {
+  load_balancer_arn = aws_lb.api.arn
+  port              = 80
+  protocol          = "HTTP"
+
+  default_action {
+    type = "redirect"
+
+    redirect {
+      port        = 443
+      protocol    = "HTTPS"
+      status_code = "HTTP_301"
+    }
+  }
+}
+
+resource "aws_lb_listener" "api" {
+  load_balancer_arn = aws_lb.api.id
+  port              = 443
+  protocol          = "HTTPS"
+  certificate_arn   = data.terraform_remote_state.core.outputs.ssl_certificate_arn
+
+  default_action {
+    target_group_arn = aws_lb_target_group.api.id
+    type             = "forward"
+  }
+}
+
+#
+# ECS Resources
+#
+resource "aws_ecs_task_definition" "api" {
+  family = "${var.project}API"
+
+  container_definitions = templatefile("${path.module}/task-definitions/api.json.tmpl", {
+    image = "quay.io/raster-foundry/granary-api:${var.image_tag}"
+
+    postgres_url      = "jdbc:postgresql://${data.terraform_remote_state.core.outputs.database_fqdn}/"
```

Comment:
This and all of the other database connection variable will need to be variables as well (FQDN, username, password, etc.).